packages/beacon-state-transition/src/cache/epochContext.ts

```diff
@@ -474,6 +485,13 @@ export class EpochContext {
         `Requesting beacon proposer for different epoch current shuffling: ${epoch} != ${this.currentShuffling.epoch}`
       );
     }
+
+    for (const cachedEpoch of this.nextEpochProposers.keys()) {
+      // Do not keep past cached future proposal duties.
+      if (cachedEpoch <= epoch) {
+        this.nextEpochProposers.delete(cachedEpoch);
```

Comment:
Why is this necessary? Once you have computed nextEpochProposers entry it can be left there